libs/features/mep/lingo.js

```diff
@@ -0,0 +1,158 @@
+/* MEP Lingo - Region-optimized content handling. See README.md for documentation. */
+import { getConfig, customFetch, queryIndexes, createTag, getUserCountry } from '../../utils/utils.js';
+
+/** Get mep-lingo context: country, localeCode, regionKey, matchingRegion */
+export function getMepLingoContext(locale) {
+  if (!locale?.prefix) {
+    return { country: null, localeCode: null, regionKey: null, matchingRegion: null };
+  }
+
+  const country = getUserCountry();
+  const config = getConfig();
+  const mapping = config.mepLingoCountryToRegion;
+
+  // Map country to region if configured (e.g., ng -> africa)
+  let regionalCountry = country;
+  if (mapping) {
+    const regionKey = Object.entries(mapping).find(
+      ([, countries]) => Array.isArray(countries) && countries.includes(country),
+    )?.[0];
+    if (regionKey) regionalCountry = regionKey;
+  }
+
+  // Derive locale code from prefix
+  const prefixParts = locale.prefix.split('/').filter(Boolean);
+  const [firstPart, secondPart] = prefixParts;
+  const hasSpecialPrefix = firstPart === 'langstore' || firstPart === 'target-preview';
+
+  let localeCode;
+  if (prefixParts.length === 0 || (hasSpecialPrefix && !secondPart)) {
+    localeCode = locale.region === 'us' ? 'en' : locale.language || 'en';
+  } else if (hasSpecialPrefix) {
+    localeCode = secondPart;
+  } else {
+    localeCode = firstPart;
+  }
```

Comment:
I think you can remove the else on line 33 by setting `let localeCode = firstPart;` here, and then the conditionals will overwrite if their conditions are met.